### PR TITLE
Document more settings for building rustc for Fuchsia

### PR DESCRIPTION
**Pull request by djkoloski**
This documents that you need to link for Fuchsia with `lld` and provides configuration settings for both `clang` and `lld`. It also adjusts the documentation for running the test suite to recommend installing to a prefix.

r? @tmandry

### CODE REVIEW

File: config-env.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Configure this environment variable to be the path to the downloaded SDK
```
Comment (tmandry) [inline config-env.sh:1]: This file shouldn't be checked in; I suggest adding it to `.git/info/exclude` or whatever the local gitignore file is.

File: src/doc/rustc/src/platform-support/fuchsia.md [hunk 2]
```diff
@@ -196,6 +196,31 @@ In `config.toml`, add:
...
+linker = "ld.lld"
+
+[target.aarch64-fuchsia]
+cc = "clang"
+cxx = "clang++"
+linker = "ld.lld"
+```
+
+Note that while `clang++` is used to link for the host platform (e.g.
+`x86_64-unknown-linux-gnu`), `lld` is used to link for Fuchsia. You may also
```
Comment (tmandry) [inline src/doc/rustc/src/platform-support/fuchsia.md:217]: I think this is already hard coded in rustc for Fuchsia.

File: src/doc/rustc/src/platform-support/fuchsia.md [hunk 3]
```diff
@@ -196,6 +196,31 @@ In `config.toml`, add:
...
+want to use LLVM's versions of `ar` and `ranlib` in a similar way:
```
Comment (tmandry) [inline src/doc/rustc/src/platform-support/fuchsia.md:218]: What would be the reason for wanting to use these?

We do it in our toolchain build to only depend on LLVM tools but I don't think outside users will care about that.

File: src/doc/rustc/src/platform-support/fuchsia.md [hunk 4]
```diff
@@ -641,20 +666,29 @@ available on the [Fuchsia devsite].
...
-1. Download of the Fuchsia SDK. Minimum supported SDK version is [9.20220726.1.1](https://chrome-infra-packages.appspot.com/p/fuchsia/sdk/core/linux-amd64/+/version:9.20220726.1.1)
+To run the Rust test suite on an emulated Fuchsia device, you must install the
+Rust compiler locally. See "[Targeting Fuchsia with a compiler built from source](#targeting-fuchsia-with-a-compiler-built-from-source)"
+for the steps to build locally. By default, the Rust compiler installs itself to
+`/usr/local` on most systems. You may want to install it to another location by
+setting a custom prefix in `config.toml`:
+
+```toml
+[install]
+prefix = "<RUST_INSTALL_PREFIX>"
```
Comment (tmandry) [inline src/doc/rustc/src/platform-support/fuchsia.md:677]: Maybe an example of using a local `install/` directory inside the source tree would be helpful? Specifically I think suggesting something that doesn't require `sudo` is a good idea :)

It looks like we use an absolute path for this one so unfortunately it still requires the user to fill something in.

File: config-env.sh (hunk 1 above)
Comment (djkoloski) [reply config-env.sh]: Removed

File: src/doc/rustc/src/platform-support/fuchsia.md (hunk 2 above)
Comment (djkoloski) [reply src/doc/rustc/src/platform-support/fuchsia.md]: It's set to use `rust-lld` by default (`compiler/rustc_target/src/spec/fuchsia_base.rs`) so we have to set `lld = true` for that to work. Added that to the build configuration.

File: src/doc/rustc/src/platform-support/fuchsia.md (hunk 3 above)
Comment (djkoloski) [reply src/doc/rustc/src/platform-support/fuchsia.md]: Removed.

File: src/doc/rustc/src/platform-support/fuchsia.md (hunk 4 above)
Comment (djkoloski) [reply src/doc/rustc/src/platform-support/fuchsia.md]: Added.

**Comment by djkoloski** [comment]
The "empty file" change for the test runner script is actually it being set to executable.

**Pull request merged**